lib/router.ex

```diff
@@ -16,14 +16,20 @@ defmodule HTTPServer.Router do
     end
   end
 
+  defp router(_req = %Request{method: "OPTIONS"}, handler) do
+    methods = get_methods(handler)
+    headers = Response.build_headers("")
+    Response.send_resp(200, "", Map.put(headers, "Allow", "#{Enum.join(methods, ", ")}"))
```

Comment:
this is a bit confusing; line 21 builds no headers, then line 22 builds headers. Maybe it should be time to look into your strategy for building headers as part of this PR?

Reply:
Line 21 builds the basic headers (content-length, content-type, and host) and line 22 adds Allow and the methods to the headers map! 

I agree though, I've been trying to brain storm how to handle the headers a bit better and I think I need to build a function that allows them to be much more modular.

Reply:
Ok I updated build headers. Currently it's been broken down into `build_headers/1` that accepts a body and `build_headers/2` that accepts body and methods. @t-keazirian and I also just worked on a solution that created `build_headers/3` that accepted a method, body and methods. I think I might need to build on that in the future when more header variations are entered into the mix, but for now method went totally unused and methods was only relevant when `method = "OPTIONS"` so I decided to break them down into two. Let me know if I should revert that though!